CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -183,12 +184,18 @@ class CameraFragment : Fragment() {
     }
 
     /** Define callback that will be triggered after a photo has been taken and saved to disk */
-    private val imageSavedListener = object : ImageCapture.OnImageSavedCallback {
-        override fun onError(imageCaptureError: Int, message: String, cause: Throwable?) {
-            Log.e(TAG, "Photo capture failed: $message", cause)
+    private val imageSavedListener: ImageCapture.OnImageSavedCallback = object : ImageCapture.OnImageSavedCallback {
+        override fun onError(exception: ImageCaptureException) {
+            Log.e(TAG, "Photo capture failed: ${exception.message}", exception)
         }
-
-        override fun onImageSaved(photoFile: File) {
+        override fun onImageSaved(outputFileResults: ImageCapture.OutputFileResults) {
+            var photoFile: File;
```

Comment:
Prefer `val` to `var` and remove semicolon. A more idiomatic Kotlin approach would be to do:
```kt
val photoFile = if (...) {
    ...
} else {
    ...
}
```